app/main.py

```diff
@@ -1,2 +1,97 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.storage_length = 8
+        self.storage = [None] * self.storage_length
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        new_item = (key, hash(key), value)
+        index = self.get_index(new_item)
+        self.storage[index] = new_item
+        if len([i for i in self.storage if i]) >= 2 * self.storage_length / 3:
+            self.resize()
+
+    def get_index(self, new_item: tuple) -> int:
```

Comment:
you can find index - index = hash(key) % self.capacity

Reply:
> you can find index - index = hash(key) % self.capacity

Yes, that's it! But I wrote method that return index in two cases:
#1: return index, if I want to chage value by existing key
`
 for item_index, item in enumerate(self.storage):
            if item:
                if item[0] == new_item[0] and item[1] == new_item[1]:
                    return item_index
`
#2 written logic, that checks is the cell empty/ If not - check next cell
`
 for item_index, item in enumerate(self.storage):
            if item:
                if item[0] == new_item[0] and item[1] == new_item[1]:
                    return item_index
`